Remove inactive tooltip rows

envex opened the pull request:
## What does this implement/fix?

To clean up the visual noise of the tooltips, we're removing inactive items from rendering in the tooltip when other series have focus.

## What do the changes look like?

| Before  | After  |
|---|---|
|![image](https://github.com/user-attachments/assets/f456d553-a5f6-4e88-8fa3-6b9dcb57df1e)|![image](https://github.com/user-attachments/assets/86a81133-7261-4a22-a0ff-86b97ed7bd48)|

 
## Storybook link

<!-- 🎩 Include links to help tophatting -->


### Before merging

- [ ] Check your changes on a variety of [browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers) and devices.

- [ ] Update the Changelog's Unreleased section with your changes.

- [ ] Update relevant documentation, tests, and Storybook.

- [ ] Make sure you're exporting any new shared Components, Types and Utilities from the top level index file of the package

michaelnesen commented on packages/polaris-viz/src/components/LineChartPredictive/utilities/renderLinearPredictiveTooltipContent.tsx:
```diff
@@ -57,11 +57,17 @@ export function renderLinearPredictiveTooltipContent(
       const isPredictiveStartKey = metadata?.startKey === activeKey;
       const isHidden = isNull || isPredictiveStartKey;
 
+      if (
+        isComparison !== true &&
+        activeColorVisionIndex !== -1 &&
+        index !== activeColorVisionIndex
+      ) {
+        return null;
+      }
```
More of a UX question, but I noticed that displaying all the series in the tooltip when there's no active index—and then switching to just one when you hover over a series can create some jarring behaviour, especially with higher series lengths. It keeps flipping between the two states and looks a bit off. Do you think there's something we could add to help minimize that effect?


https://github.com/user-attachments/assets/d79ba06d-30bc-4202-ac3e-c504beaa5c9d

envex replied:
@adamperron Thoughts?

adamperron replied:
I need some time to think about this one

envex commented:
@susiekims I've fixed your issue and @michaelnesen we've added a slight delay when resetting the tooltip content so it shouldn't jump around as much in that case.

michaelnesen reviewed (approved):
Nice! Changes look good to me 👍
